presto-hive-function-namespace/src/main/java/com/facebook/presto/hive/functions/HiveFunctionNamespaceManager.java

```diff
@@ -0,0 +1,269 @@
+import com.facebook.presto.spi.function.AlterRoutineCharacteristics;
+import com.facebook.presto.spi.function.FunctionHandle;
+import com.facebook.presto.spi.function.FunctionMetadata;
+import com.facebook.presto.spi.function.FunctionNamespaceManager;
+import com.facebook.presto.spi.function.FunctionNamespaceTransactionHandle;
+import com.facebook.presto.spi.function.ScalarFunctionImplementation;
+import com.facebook.presto.spi.function.Signature;
+import com.facebook.presto.spi.function.SqlInvokedFunction;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.google.common.collect.ImmutableList;
+import org.apache.hadoop.hive.ql.exec.UDF;
+import org.apache.hadoop.hive.ql.udf.generic.GenericUDF;
+
+import javax.inject.Inject;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Stream;
+
+import static com.facebook.presto.hive.functions.HiveFunctionErrorCode.functionNotFound;
+import static com.facebook.presto.hive.functions.HiveFunctionErrorCode.unsupportedFunctionType;
+import static com.facebook.presto.hive.functions.scalar.HiveScalarFunction.createHiveScalarFunction;
+import static com.facebook.presto.spi.StandardErrorCode.GENERIC_USER_ERROR;
+import static com.facebook.presto.spi.StandardErrorCode.NOT_SUPPORTED;
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class HiveFunctionNamespaceManager
+        implements FunctionNamespaceManager<HiveFunction>
+{
+    public static final String ID = "hive-functions";
+
+    private final ClassLoader classLoader;
+    private final LoadingCache<FunctionKey, HiveFunction> functions;
+    private final HiveFunctionRegistry hiveFunctionRegistry;
+    private final TypeManager typeManager;
+
+    @Inject
+    public HiveFunctionNamespaceManager(
+            @ForHiveFunction ClassLoader classLoader,
+            HiveFunctionRegistry hiveFunctionRegistry,
+            TypeManager typeManager)
+    {
+        this.hiveFunctionRegistry = requireNonNull(hiveFunctionRegistry, "hiveFunctionRegistry is null");
+        this.classLoader = requireNonNull(classLoader, "classLoader is null");
+        this.typeManager = requireNonNull(typeManager, "typeManager is null");
+        this.functions = CacheBuilder.newBuilder()
+                .build(CacheLoader.from(this::initializeFunction));
+    }
+
+    @Override
+    public void setBlockEncodingSerde(BlockEncodingSerde blockEncodingSerde)
+    {
+        return;
+    }
+
+    public FunctionNamespaceTransactionHandle beginTransaction()
```

Comment:
Maybe add comment at the top that this function namespace manager does not support transaction. As long as the functions are static, not supporting transaction is fine.